src/odemis/acq/stitching/_registrar.py

```diff
@@ -580,8 +581,18 @@ def _insert_tile_to_grid(self, tile):
                     self.tiles[i].append(None)
                     self.shifts_hor[i].append(None)
                     self.shifts_ver[i].append(None)
-        else:
+        elif abs(hor_diff) == 0 and abs(ver_diff) == 0:
             raise ValueError("Cannot insert multiple tiles at the same position.")
+        elif abs(ver_diff) == abs(hor_diff):
+            # Pure diagonal shift. Via the previous if statements is implied that hor_diff and ver_diff are not zero.
```

Comment:
Looks good. Maybe we can make the comments a bit clearer. For example here, I would put something like
```# Purely diagonal shift. In practice, this will never happen because the smallest difference will be picked up by one of the previous if-statements. This exception is likely to be triggered only in artificial tests.```

TODO is not needed in my opinion. The only way to deal with it I can think of is to add a small value to one of the shifts, so they are not exactly equal anymore. However, we probably prefer to get an error in this case (especially since this registrar is not supposed to handle diagonal moves anyway).